### DISCUSSION TITLE
YOLOv8 CLI is saving video, even though save=False should be default

### DISCUSSION THREAD

**Discussion by trevorstr** [ultralytics/ultralytics#9957] [General]
I am running the YOLOv8 CLI tool from the `ultralytics/ultralytics:latest` container image, and pointing it to an RTSP stream for ingestion. I need the stdout from the CLI with detections, but I do not need access to any of the video or image (frames) data.

According to [the documentation](https://docs.ultralytics.com/modes/predict/#inference-arguments), the default `save=False` parameter is set, but it's still outputting an AVI file into the ./runs/detect/predict/`.

Is this a bug? How do I tell the CLI not to output the video that it's ingesting?

It's filling up my disk, and I don't need to save any data besides the Docker logs.

<img width="884" alt="Screenshot 2024-04-10 at 11 19 50 AM" src="https://github.com/ultralytics/ultralytics/assets/147647647/14933d53-09f2-47d3-81d7-ef60f282b1c4">

**Comment by pderrenger** [comment]
Hello!

It seems like there might be a slight confusion or an unexpected situation occurring with the CLI tool. Indeed, the default for `save=False` should prevent any video files from being saved during the prediction process. However, if you're experiencing differently, a workaround is to explicitly set the `save` parameter to `False` in your command, just to ensure this setting is enforced. Here is how you can specify it:

```bash
yolo detect predict source='rtsp://your_stream' model=yolov8n.pt save=False
```

If specifying `save=False` explicitly does not resolve the issue, it's possible you've encountered a bug or an unexpected behavior specific to the Docker image version you're using.

In such a case, reporting this with detailed steps to reproduce on the [Ultralytics YOLO GitHub issues](https://github.com/ultralytics/ultralytics/issues) would be greatly appreciated, as it helps improve the tool for everyone. 🛠️

Meanwhile, as a temporary measure, you might consider including a step in your workflow to clear out the `./runs/detect/predict/` directory periodically or after each run to manage disk space.

Thanks for bringing this to our attention!

**Comment by trevorstr** [comment]
> workaround is to explicitly set the save parameter to False in your command

I just tried that, and then used `docker exec` to spawn an interactive shell inside the container. It seems that the `./runs` directory is now not being created, as expected.

This seems like a bug, either a doc or functional. The documentation states that the default is to not save media locally, but apparently that's not the case. I filed a bug report here ➡️  https://github.com/ultralytics/ultralytics/issues/9972 

This is the command I'm using to launch my containers, from the unmodified `ultralytics/ultralytics:latest` image. It's much easier to use the stock image rather than creating a custom one.

```
docker run --gpus=all --restart=always -it \
  --detach ultralytics/ultralytics \
  bash -c 'curl -LO https://github.com/ultralytics/assets/releases/download/v8.1.0/yolov8l.pt; yolo predict source=rtsp://user:pass@ip_address imgsz=1920 model=yolov8l.pt conf=0.8 save=False'
```

Thanks for your response! You guys are awesome.

**Reply by pderrenger** [reply]
Hi there! 🙌 Thanks for bringing this to our attention, and we really appreciate your patience and detailed feedback. It seems like you've stumbled upon a discrepancy, and we're on it.

The command line you're using is pretty solid. However, when setting the `save` parameter via CLI, please ensure it's written in lowercase `false`, as booleans are case-sensitive in most command-line environments.

Try adjusting your command as follows:
```bash
docker run --gpus=all --restart=always -it \
  --detach ultralytics/ultralytics \
  bash -c 'curl -LO https://github.com/ultralytics/assets/releases/download/v8.1.0/yolov8l.pt; yolo predict source=rtsp://user:pass @trevorstr imgsz=1920 model=yolov8l.pt conf=0.8 save=false'
```

This should prevent the saving of video files and align with your expectations. We'll definitely review the documentation and ensure it's clear and accurate for users to avoid any confusion moving forward.

Thanks for using our model, and do not hesitate to reach out again if you encounter any more issues or have further queries! Keep an eye on the issue you filed for any updates on this matter. Cheers! 🚀
